private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -791,6 +793,92 @@ export class TemplateServiceClient {
     return { success: true, result: resultTemplates };
   }
 
+  /**
+   * @public
+   * Get entry point.
+   * Returns the specified template with the data bound into the template JSON.
+   * If a version is passed in, it will update the template JSON of that version.
+   * If not version is passed, it will update the latest version.
+   * @param {string} templateId - unique template id
+   * @param {JSON} data - data json to bound with template json
+   * @param {string} version - version number, used with templateId
+   * @param {boolean} isClient - used to ignore updating the hit number on a template
+   * @returns Promise as valid json
+   */
+  public async bindData(
+    token: string,
+    templateId: string,
+    data: JSON,
+    version?: string,
+    isClient?: boolean,
+
+  ): Promise<JSONResponse<ITemplate[]>> {
+    let authCheck = this._checkAuthenticated(token);
+    if (!authCheck.success) {
+      return authCheck;
+    }
+    let authId = this.authProvider.getAuthIDFromToken(token || this.authProvider.token);
+    let userResponse = await this._getUser(authId);
+    if (!userResponse.success || !userResponse.result || userResponse.result.length === 0) {
+      return { success: false, errorMessage: userResponse.errorMessage };
+    }
+
+    let userId = userResponse.result![0]._id;
+    const templateQuery: Partial<ITemplate> = {
+      _id: templateId,
+    };
+
+    let response = await this.storageProvider.getTemplates(templateQuery, undefined, undefined);
+
+    if (!response.success || !response.result) return response;
+
+    let templates: ITemplate[] = response.result;
+
+    if (templateId && templates.length > 0) {
```

Comment:
templateid must be defined based on the function params, we can probably add a check that the templates.length === 0 before we proceed (since only one template object should be returned from the database because we pass in the id)

Reply:
Do you mean check that templates.length === 1?

Reply:
Oh yep my bad check templates.length === 1!